test/CQRS/LeanCode.CQRS.AspNetCore.Tests/Local/MiddlewareBasedExecutorIntegrationTests.cs

```diff
@@ -0,0 +1,143 @@
+using System.Net.Http.Json;
+using System.Security.Claims;
+using FluentAssertions;
+using LeanCode.Components;
+using LeanCode.Contracts;
+using LeanCode.CQRS.AspNetCore.Local;
+using LeanCode.CQRS.Execution;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.TestHost;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+
+namespace LeanCode.CQRS.AspNetCore.Tests.Local;
+
+public class MiddlewareBasedExecutorIntegrationTests : IDisposable, IAsyncLifetime
```

Comment:
Wouldn't using `WebApplicationFactory` over raw host with `IAsyncLifetime` be more convenient?

Reply:
I would still need to defined `Program` somewhere, I don't think it will result in substantially easier code.